internal/ioutil/io.go

```diff
@@ -37,31 +37,14 @@ func (fn CloserFunc) Close() error {
 // The copied content is verified against the size and the digest.
 func CopyBuffer(dst io.Writer, src io.Reader, buf []byte, desc ocispec.Descriptor) error {
 	// verify while copying
-	verifier := desc.Digest.Verifier()
-	lr := io.LimitReader(src, desc.Size)
-	mw := io.MultiWriter(verifier, dst)
-
-	if _, err := io.CopyBuffer(mw, lr, buf); err != nil {
+	vr := content.NewVerifyReader(src, desc)
+	if _, err := io.CopyBuffer(dst, vr, buf); err != nil {
 		return fmt.Errorf("copy failed: %w", err)
 	}
-
-	if !verifier.Verified() {
-		return errors.New("digest verification failed")
+	if err := vr.Verify(); err != nil {
+		return err
 	}
-
-	return EnsureEOF(lr)
-}
-
-// EnsureEOF ensures the read operation ends with an EOF and no
-// trailing data is present.
-func EnsureEOF(r io.Reader) error {
-	var peek [1]byte
-	_, err := io.ReadFull(r, peek[:])
-	if err != io.EOF {
-		return errors.New("trailing data")
-	}
-
-	return nil
+	return content.EnsureEOF(vr)
```

Comment:
Do we still need this? EOF is already ensured in `Verify()`.

Reply:
removed it.